File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/impl/nms/bridges/packets/SurfBukkitNmsPacketBridgesImpl.java
package dev.slne.surf.surfapi.bukkit.server.impl.nms.bridges.packets;

import dev.slne.surf.surfapi.bukkit.api.nms.bridges.packets.SurfBukkitNmsPacketBridges;
import dev.slne.surf.surfapi.bukkit.server.impl.nms.bridges.packets.block.SurfBukkitNmsBlockPacketsImpl;
import dev.slne.surf.surfapi.bukkit.server.impl.nms.bridges.packets.entity.SurfBukkitNmsSpawnPacketsImpl;
import dev.slne.surf.surfapi.bukkit.server.impl.nms.bridges.packets.player.SurfBukkitNmsPlayerPacketsImpl;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public final class SurfBukkitNmsPacketBridgesImpl implements SurfBukkitNmsPacketBridges {

  private final SurfBukkitNmsSpawnPacketsImpl spawnPackets = new SurfBukkitNmsSpawnPacketsImpl();
  private final SurfBukkitNmsBlockPacketsImpl blockPackets = new SurfBukkitNmsBlockPacketsImpl();
  private final SurfBukkitNmsPlayerPacketsImpl playerPackets = new SurfBukkitNmsPlayerPacketsImpl();

  @Override
  public PacketOperationImpl createEmptyPacketOperation() {
    return PacketOperationImpl.empty();
  }
}

File: surf-api-bukkit/surf-api-bukkit-server/src/main/kotlin/dev/slne/surf/surfapi/bukkit/server/impl/nms/bridges/packets/SurfBukkitNmsPacketBridgesImpl.kt
package dev.slne.surf.surfapi.bukkit.server.impl.nms.bridges.packets

import com.google.auto.service.AutoService
import dev.slne.surf.surfapi.bukkit.api.nms.NmsUseWithCaution
import dev.slne.surf.surfapi.bukkit.api.nms.bridges.packets.SurfBukkitNmsPacketBridges
import dev.slne.surf.surfapi.core.api.util.checkInstantiationByServiceLoader

@NmsUseWithCaution
@AutoService(SurfBukkitNmsPacketBridges::class)
class SurfBukkitNmsPacketBridgesImpl : SurfBukkitNmsPacketBridges {
    init {
        checkInstantiationByServiceLoader()
    }

    override fun createEmptyPacketOperation(): PacketOperationImpl {
        return PacketOperationImpl.empty()
    }
}
